client/src/tests/qNa.helper.test.js

```diff
@@ -2,33 +2,178 @@ import { expect, jest, test, describe, beforeEach, afterEach } from '@jest/globa
+    })
+    test('sellers should be sorted by helpfulness', () => {
+      let sorted;
+      sortedAns = helper.sortAnswers(newQuestionsWithAnsArr[0].answers)
+        .filter((answer) => {
+          if (answer.answerer_name.toLowerCase() ==='seller') {
+            return helper.sortAnswers(newQuestionsWithAnsArr[0].answers)
+          }
+        })
+
+      sortedAns.forEach((answer, i) => {
+        if (sortedAns[i+1]) {
+          if (answer.helpfulness >= sortedAns[i+1].helpfulness) {
+            sorted = true
+          } else {
+            sorted = false
+          }
+        }
+        expect(sorted).toBe(true)
+      })
+    })
+    test('users should be sorted by helpfulness', () => {
+      let sorted;
+      sortedAns = helper.sortAnswers(newQuestionsWithAnsArr[0].answers)
+        .filter((answer) => {
+          if (answer.answerer_name.toLowerCase() !=='seller') {
+            return helper.sortAnswers(newQuestionsWithAnsArr[0].answers)
+          }
+        })
 
+      sortedAns.forEach((answer, i) => {
+        if (sortedAns[i+1]) {
+          if (answer.helpfulness >= sortedAns[i+1].helpfulness) {
+            sorted = true
+          } else {
+            sorted = false
+          }
+        }
+        expect(sorted).toBe(true)
+      })
     })
-    test('', () => {
-      const mock = jest.fn()
-      const bound = mock.bind(helper)
+  })
 
-      bound()
-      // bound.sortQuestions(questions)
-      let sortQuestions = mock.mock.instances[0].sortQuestions
-      sortQuestions(questions)
-      jest.spyOn(mock.mock.instances[0], 'sortQuestions')
+  describe('filterSearchInput', () => {
+    test('should return a new questions results array, that include search term', ()=> {
+      let questions= testData.data();;
+      let value = '123'
+      let value2='photos for all'
+      let newQuestions = helper.filterSearchInput(questions, value);
+      let newQuestions2 = helper.filterSearchInput(questions, value2);
+      newQuestions.forEach(question => {
+        expect(question.question_body.includes(value)).toBe(true)
+      })
+      newQuestions2.forEach(question => {
+        expect(question.question_body.includes(value)).toBe(true)
+      })
+    })
+  })
 
-      // console.log(mock.mock.instances[0].sortQuestions.mock)
-      expect(mock.mock.instances[0].sortQuestions).toBeCalledTimes(1)
+  describe('moreAnsweredQButtonDisplay', () => {
+    let clickCount;
+    let index;
+    test('it should return true, when input click count equals input index', () => {
+      clickCount =3
+      index = 3
+      expect(helper.moreAnsweredQButtonDisplay(clickCount, index)).toBe(true)
+    })
+    test('it should return true, when clickCount - 1 equals index',  () => {
+      clickCount = 5
+      index = 4
+      expect(helper.moreAnsweredQButtonDisplay(clickCount, index)).toBe(true)
+
+    })
+    test('it should return false, when clickCount does not equal index', () => {
```

Comment:
This test following the test above seems confusing. On line 124, it says it should return if clickCount - 1 equals index. In this test, clickCount - 1 also equals index, so I would think it should return true? Could be that you are testing specific scenario where index = 0 or something in second test, but a more descriptive test name in that case would help